src/components/GameCard.tsx

```diff
@@ -22,12 +25,34 @@ interface GameCardProps {
     notes?: string;
     platformsOwnedOn?: Array<string>;
     dateAdded?: Date;
+}
 
+interface LibraryOption {
+    label: string,
+    value: string
 }
 
 export function GameCard(gameCardProps : GameCardProps ) {
-    // will eventually be an API function
-    const addToLibrary = () => true;
+    const [libraryOptions, setAllLibraryOptions] = useState<Array<LibraryOption>>([])
+    const [selectedLibrary, setSelectedLibrary] = useState("")
+
+    useEffect(() => {
+        const fetchAllLibraries = async () => {
```

Comment:
Do we want to fetch all libraries on page load, or only when a user clicks the dropdown?
Tradeoff is slower page load vs. slower dropdown behavior, but I feel like it might be worth choosing the slower dropdown since that will only affect a subset of users who visit the page as opposed to all

Reply:
yeah i agree, changing